sass/susy/_parser.scss

```diff
@@ -30,38 +34,68 @@
     'wide': 'spread',
     'wider': 'spread',
   );
-  $span-error: 'Not able to determine column-span from `#{$shorthand}`';
-  $parse-error: 'Unknown span property:';
-  $parsed: ();
+
+  $return: ();
   $span: null;
+  $columns: null;
+
+  $of: false;
+  $next: false;
 
   @for $i from 1 through length($shorthand) {
     $item: nth($shorthand, $i);
     $type: type-of($item);
 
-    @if ($type == 'string') {
-      @if map-has-key($options, $item) {
-        $item: (map-get($options, $item): $item);
-        $parsed: map-merge($parsed, $item);
-      } @else if not $span {
-        @error $span-error;
-      } @else if ($item == 'at') {
-        $next: nth($shorthand, $i + 1);
-        $parsed: map-merge($parsed, ('location': $next));
-      } @else if ($item == 'of') {
-        $next: nth($shorthand, $i + 1);
-        $parsed: map-merge($parsed, ('columns': $next));
+    @if $next {
+      @if $next == 'column-width' {
+        @if $type == 'number' and not unitless($item) {
+          @if $columns {
+            $columns: join($columns, ('x' $item));
```

Comment:
To explain this change: I just keep looking for the best way to handle multi-item arguments in the parser. We have several: `of $n`, `at $n`, and `x $n`. In the old version we just grabbed the next value inline, and then had to check the previous value on every loop to make sure we hadn't already used the current value. This seemed more robust to me, because we just let each iteration of the loop do it's own thing.